lib/ccnet/ccnet.js

```diff
@@ -0,0 +1,208 @@
+      this._connect()
+
+      const t0 = Date.now()
+      const denominationsInterval = setInterval(() => {
+        if (this.hasDenominations()) {
+          clearInterval(denominationsInterval)
+          return cb()
+        }
+
+        if (Date.now() - t0 > 5000) {
+          clearInterval(denominationsInterval)
+          cb(new Error('Timeout waiting for denominations'))
+        }
+      }, 500)
+    })
+    .catch(cb)
+}
+
+Ccnet.prototype.close = function (cb) {
+  this._stopPolling()
+  rs232.close(function (err) {
+    if (err) console.log(err)
+    cb(err)
+  })
+}
+
+Ccnet.prototype.getBillsToEnable = function () {
+  // at this point this enables every bill available on the firmware
+  // take fiatCode into account once we do that in id003 as well
+  const billsBuffer = Buffer.alloc(3)
+  const billsToEnable = _.keys(this._denominations())
+
+  billsToEnable.forEach(it => {
+    writeBit(billsBuffer, 2 - Math.floor(it / 8), it % 8, 1)
+  })
+
+  return Buffer.concat([billsBuffer, billsBuffer])
+}
+
+Ccnet.prototype.enable = function () {
+  const billsBuffer = this.getBillsToEnable()
+
+  const command = Buffer.concat([Buffer.from(commands.ENABLE_BILL_TYPES), billsBuffer])
+  this._send(command)
+}
+
+Ccnet.prototype.disable = function () {
+  const command = Buffer.concat([Buffer.from(commands.ENABLE_BILL_TYPES), Buffer.alloc(6)])
+  this._send(command)
+}
+
+Ccnet.prototype.stack = function () {
+  this._send(commands.STACK)
+}
+
+Ccnet.prototype.reject = function () {
+  this._send(commands.RETURN)
+}
+
+Ccnet.prototype.lowestBill = function (fiat) {
+  const bills = _.values(this._denominations())
+  const filtered = bills.filter(bill => fiat.lte(bill))
+  if (_.isEmpty(filtered)) return BN(Infinity)
+  return BN(_.min(filtered))
+}
+
+Ccnet.prototype.highestBill = function (fiat) {
+  const bills = _.values(this._denominations())
+  const filtered = bills.filter(bill => fiat.gte(bill))
+  if (_.isEmpty(filtered)) return BN(-Infinity)
+  return BN(_.max(filtered))
+}
+
+Ccnet.prototype.hasDenominations = function () {
+  return this._denominations() !== null
+}
+
+Ccnet.prototype._denominations = function () {
+  return this.denominations
+}
+
+Ccnet.prototype._send = function (command) {
+  this._stopPolling()
+  setTimeout(() => {
+    rs232.request(command)
+    this._startPolling()
+  }, POLLING_INTERVAL)
+}
+
+Ccnet.prototype._connect = function () {
+  statesFsm.handle('CONNECT')
+}
+
+Ccnet.prototype._startPolling = function () {
+  this.pollingInterval = setInterval(() => {
+    rs232.request(commands.POLL)
+  }, POLLING_INTERVAL)
+}
+
+Ccnet.prototype._stopPolling = function () {
+  clearInterval(this.pollingInterval)
+}
+
+Ccnet.prototype._handleResponse = function (response, command) {
+  if (command === commands.GET_BILL_TABLE) {
+    return this._billTable(response)
+  }
+
+  return statesFsm.handle(respMap[response[0]], response.slice(1), command)
+}
+
+Ccnet.prototype._billTable = function (data) {
```

Comment:
Since this creates side effects and isn't a pure function, describe side effects in a comment.